File: Zestawy Zadań/Python/1.1 - Operatory wyjścia - wejścia/Zestaw 2 zadania/1.7.py
//Napisz program, który:
//a.Pobiera trzy znaki
//b.W kolejnych wierszach wypisuje wszystkie możliwe permutacje tych znaków.Przykładowo
//dla znaków ‘K’, ‘O’ i ‘T’ program wypisuje(w dowolnej kolejności) : „KOT”, „KTO”, „OKT”,
//„OTK”, „TOK” i „TKO”.


from itertools import permutations

def print_permutations(chars):
    perms = permutations(chars)
    for perm in perms:
        print(''.join(perm))

char1 = input("Podaj pierwszy znak: ")
char2 = input("Podaj drugi znak: ")
char3 = input("Podaj trzeci znak: ")

chars = [char1, char2, char3]

print("Wszystkie możliwe permutacje:")
print_permutations(chars)

File: Zestawy Zadań/C++/1 - Operatory wyjscia-wejscia/Zestaw 2 zadania/1.7.cpp
//Napisz program, który:
//a.Pobiera trzy znaki
//b.W kolejnych wierszach wypisuje wszystkie możliwe permutacje tych znaków.Przykładowo
//dla znaków ‘K’, ‘O’ i ‘T’ program wypisuje(w dowolnej kolejności) : „KOT”, „KTO”, „OKT”,
//„OTK”, „TOK” i „TKO”.

#include <iostream>
#include <algorithm>

using namespace std;

int main() {
    string znaki;
    cout << "Podaj trzy znaki (bez spacji): ";
    cin >> znaki;

    sort(znaki.begin(), znaki.end());
    do {
        cout << znaki << "\n";
    } while (next_permutation(znaki.begin(), znaki.end()));

    return 0;
}
